### PR TITLE
add additional lucene index configs

### PR DESCRIPTION
**Pull request by itschrispeck**
Adds two configs for Lucene backed text index, both can be used to optimize for index writing time
1. luceneUseCompoundFile: [compound file format](https://lucene.apache.org/core/8_1_1/core/org/apache/lucene/index/IndexWriterConfig.html#setUseCompoundFile-boolean-)
2. luceneMaxBufferSizeMB: [max buffer size](https://lucene.apache.org/core/8_1_1/core/org/apache/lucene/index/IndexWriterConfig.html#setRAMBufferSizeMB-double-)

Internal testing showed tuning these can improve realtime -> offline segment build time by ~30% (~50% for the 'lucene portion' when subtracting out build time for a table w/o text index). Some sample settings are shown below, collected by creating identical (minus lucene configs) tables in the same cluster. Build time is visualized via ingestion delay metrics: 

image

### CODE REVIEW

File: pinot-segment-spi/src/main/java/org/apache/pinot/segment/spi/index/TextIndexConfig.java [hunk 1]
```diff
@@ -32,15 +32,20 @@
 
 
 public class TextIndexConfig extends IndexConfig {
-  public static final TextIndexConfig DISABLED = new TextIndexConfig(true, null, null, false, false,
-      Collections.emptyList(), Collections.emptyList());
+  private static final int LUCENE_INDEX_DEFAULT_MAX_BUFFER_SIZE_MB = 500;
```
Comment (Jackie-Jiang) [inline pinot-segment-spi/src/main/java/org/apache/pinot/segment/spi/index/TextIndexConfig.java:35]: Let's also add a constant for:
```
  private static final int LUCENE_INDEX_DEFAULT_USE_COMPOUND_FILE = true;
```

File: pinot-segment-spi/src/main/java/org/apache/pinot/segment/spi/index/TextIndexConfig.java [hunk 2]
```diff
@@ -50,14 +55,19 @@ public TextIndexConfig(
...
+      @JsonProperty("luceneUseCompoundFile") Boolean luceneUseCompoundFile,
+      @JsonProperty("luceneMaxBufferSizeMB") Integer luceneMaxBufferSizeMB) {
     super(disabled);
     _fstType = fstType;
     _rawValueForTextIndex = rawValueForTextIndex;
     _enableQueryCache = enableQueryCache;
     _useANDForMultiTermQueries = useANDForMultiTermQueries;
     _stopWordsInclude = stopWordsInclude;
     _stopWordsExclude = stopWordsExclude;
+    _luceneUseCompoundFile = luceneUseCompoundFile == null || luceneUseCompoundFile; // default to useCompoundFile
```
Comment (Jackie-Jiang) [inline pinot-segment-spi/src/main/java/org/apache/pinot/segment/spi/index/TextIndexConfig.java:68]:
```suggestion
    _luceneUseCompoundFile = luceneUseCompoundFile == null ? LUCENE_INDEX_DEFAULT_USE_COMPOUND_FILE : luceneUseCompoundFile;
```

File: pinot-segment-spi/src/main/java/org/apache/pinot/segment/spi/index/TextIndexConfig.java [hunk 3]
```diff
@@ -99,6 +123,8 @@ public static abstract class AbstractBuilder {
     protected boolean _useANDForMultiTermQueries = true;
     protected List<String> _stopWordsInclude = new ArrayList<>();
     protected List<String> _stopWordsExclude = new ArrayList<>();
+    protected boolean _luceneUseCompoundFile = true;
```
Comment (Jackie-Jiang) [inline pinot-segment-spi/src/main/java/org/apache/pinot/segment/spi/index/TextIndexConfig.java:126]:
```suggestion
    protected boolean _luceneUseCompoundFile = LUCENE_INDEX_DEFAULT_USE_COMPOUND_FILE;
```

**Review by Jackie-Jiang — APPROVED** [review]
LGTM otherwise, well done!

File: pinot-segment-spi/src/main/java/org/apache/pinot/segment/spi/index/TextIndexConfig.java (hunk 1 above)
Comment (itschrispeck) [reply pinot-segment-spi/src/main/java/org/apache/pinot/segment/spi/index/TextIndexConfig.java:35]: 👍 thanks for the review!

**Comment by Jackie-Jiang** [comment]
Can you also help update the Pinot doc (this [page](https://docs.pinot.apache.org/basics/indexing/text-search-support)) for the new configs?

**Pull request merged**